pkg/cmd/release/create/create.go

```diff
@@ -382,15 +399,20 @@ type PackageVersionOverride struct {
 // This is the inverse of ParsePackageOverrideString
 func (p *PackageVersionOverride) ToPackageOverrideString() string {
 	components := make([]string, 0, 3)
-	if p.PackageReferenceName != "" {
-		components = append(components, p.PackageReferenceName)
-	}
+
+	// stepNameOrPackageID always comes first if we have it
```

Comment:
I had a bug here; I'd mis-interpreted the C# code in the server as being `packageReference:stepName:version` but it was actually `stepName:packageReferenceName:version`. See unit test changes etc